### PR TITLE
feat: workload generator support create contentFrom based file

### LINKED ISSUE #673: Feat: support using ContentFrom in FileSpec
### What would you like to be added?

Support using `ContentFrom` in `FileSpec` to create ConfigMap and volume & volumeMount in Workload.

### Why is this needed?

Now if using `ContentFrom` in `FileSpec`, there will be panic.

### PR DESCRIPTION
**Pull request by adohe**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
workload generator support create contentFrom based file
#### Which issue(s) this PR fixes:

Fixes #673 

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., design docs, usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/modules/generators/workload/workload_generator.go [hunk 1]
```diff
@@ -440,3 +458,41 @@ func completeBaseWorkload(base *workload.Base, config apiv1.GenericConfig) error
...
+
+	u, err := url.Parse(s)
+	if err != nil {
+		return result, false, err
+	}
+
+	result.Name = u.Host
+	result.Key, _, _ = strings.Cut(strings.TrimPrefix(u.Path, "/"), "/")
+
+	/*q := u.Query()
```
Comment (SparkYuan) [inline pkg/modules/generators/workload/workload_generator.go:484]: Are these commented lines still necessary? Can they be deleted?

File: pkg/modules/generators/workload/workload_generator.go [hunk 2]
```diff
@@ -369,8 +369,26 @@ func handleFileCreation(c container.Container, uniqueAppName, containerName stri
 		}
 
 		if v.ContentFrom != "" {
-			// TODO: support the creation of the file content from a reference source.
-			panic("not supported the creation the file content from a reference source")
+			sec, ok, err := parseSecretReference(v.ContentFrom)
```
Comment (SparkYuan) [inline pkg/modules/generators/workload/workload_generator.go:372]: Adding an example in the comments may help readers understand the codes better. The test cases in the unit tests are quite appropriate.

Comment (adohe) [reply pkg/modules/generators/workload/workload_generator.go:372]: just add comments to `parseSecretReference` function.

**Review by liu-hm19 — APPROVED** [review]
lgtm

**Pull request merged**